packages/compass-indexes/src/components/search-indexes-table/search-indexes-table.tsx

```diff
@@ -174,11 +178,15 @@ export const SearchIndexesTable: React.FunctionComponent<
   onPollIndexes,
 }) => {
   useEffect(() => {
+    if (!isWritable) {
+      return noop;
+    }
+
     const id = setInterval(onPollIndexes, POLLING_INTERVAL);
     return () => {
       clearInterval(id);
     };
-  }, [onPollIndexes]);
+  }, [onPollIndexes, isWritable]);
```

Comment:
This is business logic that should probably be part of the store action, not component

Reply:
If we do this in the action, we will have an interval that just does nothing being called every 5s. If the component is responsible of polling every 5s, I think it should be also responsible of not polling whenever the circumstances do not allow polling.

WDYT?

Reply:
You are spreading an important part of business logic in two places, the result of this is no single source of truth for how polling should behave. Consider this: what if I want to add polling in some other place? How can I know that there is additional logic only in the component that I need to copy to the new place?

Reply:
> I want to avoid is to have the polling in the component, but additional logic to the polling in the store, spreading something that should be together in both the UI and the store.

I'm not sure I follow this part completely: you already spread it between two places, I'm just asking not to diverge the logic even more.

Feel free to keep it as-is if you feel strong about doing it that way.

Reply:
I did the change you suggested after discussing it with Basit 😄 . I think we can do it in two steps:

1. We move the logic of the "offline mode" to the action, where it should be, as you mentioned.
2. We create, either a high-order component, or some redux action, that handles the setInterval, instead of having it in the search table, so it can be reused.

This PR will do only 1. The 2. needs a little bit more thought and we can do it when we need to poll from any other place.